enhancements/ingress/custom-route-configuration.md

```diff
@@ -0,0 +1,262 @@
+    // If a ComponentRoute is created with a namespace,name tuple that does not match status, that piece of config will
+    // not have an effect.  If an operator later reads the field, it will eventually (but not necessarily immediately)
+    // honor the pre-existing spec values.
+    ComponentRoutes []ComponentRouteSpec
+}
+
+type IngressStatus struct {
+    // other fields
+
+    // ComponentRoutes is where participating operators place the current route status for routes which the cluster-admin
+    // can customize hostnames and serving certificates.
+    // How the operator uses that serving certificate is up to the individual operator.
+    // An operator that creates entries in this slice should clean them up during removal (if it can be removed).
+    // An operator must also handle the case of deleted status without churn.
+    ComponentRoutes []ComponentRouteStatus
+}
+
+type ComponentRouteSpec struct{
+    // namespace is the namespace of the route to customize.  It must be a real namespace.  Using an actual namespace
+    // ensures that no two components will conflict and the same component can be installed multiple times.
+    Namespace string
+    // name is the *logical* name of the route to customize.  It does not have to be the actual name of a route resource.
+    // Keep in mind that this is your API for users to customize.  You could later rename the route, but you cannot rename
+    // this name.
+    Name string
+    // Hostname is the host name that a cluster-admin wants to specify
+    Hostname string
+    // ServingCertKeyPairSecret is a reference to a secret in namespace/openshift-config that is a kubernetes tls secret.
+    // The serving cert/key pair must match and will be used by the operator to fulfill the intent of serving with this name.
+    // That means it could be embedded into the route or used by the operand directly.
+    // Operands should take care to ensure that if they use passthrough termination, they properly use SNI to allow service
+    // DNS access to continue to function correctly.
+    // SANs in the certificate are ignored, but SNI can be used to make operator managed certificates (like internal load balancers
+    // and service serving certificates) serve correctly.
+    ServingCertKeyPairSecret SecretNameReference
+
+    // possible future, we could add a set of SNI mappings.  I suspect most operators would not properly handle it today.
+}
+
+type ComponentRouteStatus struct{
+    // namespace is the namespace of the route to customize.  It must be a real namespace.  Using an actual namespace
+    // ensures that no two components will conflict and the same component can be installed multiple times.
+    Namespace string
+    // name is the *logical* name of the route to customize.  It does not have to be the actual name of a route resource.
+    // Keep in mind that this is your API for users to customize.  You could later rename the route, but you cannot rename
```

Comment:
misplaced in API docs. You don't talk to the dev here, but the admin.

Reply:
> misplaced in API docs. You don't talk to the dev here, but the admin.

I'm sure it will work itself out in the API PR.  Tihs expresses the intent pretty clearly.